app/services/action_handlers/pull_request.rb

```diff
@@ -26,27 +26,16 @@ def closed
     end
 
     def review_request_removed
-      pr_data = @payload['pull_request']
-      reviewers_github_id = pr_data['requested_reviewers'].map { |reviewer| reviewer['id'] }
-      remove_review_requests(reviewers_github_id)
+      removed_reviewer = find_or_create_user(@payload['requested_reviewer'])
+      @entity.review_requests.find_by(reviewer: removed_reviewer, state: 'active')
+                             &.removed!
     end
 
     def review_requested
       pr_data = @payload['pull_request']
+      reviewer = find_or_create_user(@payload['requested_reviewer'])
       owner = find_or_create_user(pr_data['user'])
-      pr_data['requested_reviewers'].each do |raw_reviewer|
-        reviewer = find_or_create_user(raw_reviewer)
-        @entity.review_requests.create!(owner: owner, reviewer: reviewer, state: 'active')
-      rescue ActiveRecord::RecordInvalid => e
-        raise unless e.message == 'Validation failed: Pull request has already been taken'
-      end
-    end
-
-    def remove_review_requests(reviewers)
-      @entity.review_requests
-             .includes(:owner)
-             .where.not(reviewer: User.where(github_id: reviewers))
-             .each(&:removed!)
+      @entity.review_requests.create(reviewer: reviewer, owner: owner, state: 'active').persisted?
```

Comment:
Why are we calling `persisted?`?

Reply:
The method `create` by default returns the object, even if it no saved, so with `persisted?` we can be sure if the record was saved.

Reply:
Also, it does not hit the database twice.

Reply:
But do we use the result of the method?

Reply:
Rubocop finds using create without the exclamation mark a bad practice, so, i found this is an alternative for that, otherwise I would have to use a rescue to catch the exception. I guess that is a bad practice to don't check if the record was or not saved.

Reply:
Sorry to reiterate on this issue, but could it be the case that it's the other way around?

Reply:
Yep, it says that you have to use the exclamation mark, but I searched that specific error and I found that you can do the same by adding the persisted? at the end